interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -173,11 +190,19 @@ public void tearDown() throws Exception {
     }
   }
 
-  protected abstract ManagedChannel createChannel();
+  protected abstract ManagedChannel createChannel(CensusContextFactory censusFactory);
+
+  protected boolean metricsExpected() {
+    return true;
+  }
 
   @Test(timeout = 10000)
   public void emptyUnary() throws Exception {
-    assertEquals(Empty.getDefaultInstance(), blockingStub.emptyCall(Empty.getDefaultInstance()));
+    assertEquals(EMPTY, blockingStub.emptyCall(EMPTY));
+
+    assertMetrics("grpc.testing.TestService/EmptyCall", Status.Code.OK,
```

Comment:
As discussed offline, checking metrics for each test is a bit of overkill and just makes the tests harder to maintain. These should be considered more "integration" tests that just test that the pieces are actually connected.

It looks to me that just testing with `largeUnary()` would be enough. It seems the other cases should be covered via normal unit tests. It's hard to say, but you could also maybe simplify `assertMetricsForOneSide()` to just spot-check that it seems "stats is on" instead of verifying all the bits.